gst/tensor_mux/gsttensormux.c

```diff
@@ -400,6 +399,17 @@ gst_tensor_mux_collected (GstCollectPads * pads, GstTensorMux * tensor_mux)
   if (!tensor_mux->negotiated) {
     GstCaps *newcaps;
 
+    if (GST_IS_BUFFER (tensors_buf)) {
+      tensor_mux->tensors_config.info.num_tensors =
+          gst_buffer_n_memory (tensors_buf);
```

Comment:
Is this safe? Does GSTBUFFER allow arbitrary number of memory chunks? (I don't think so)

Reply:
@myungjoo Although GstBuffer can hold arbitrary number of memory chunks, NNStreamer works under the assumption that each GstBuffer for input and output has same number of memory chunks as the number of tensors included in those input and output streams. Please see the following examples:
    
https://github.com/nnsuite/nnstreamer/blob/a6d3314c0ad3781bfa00cbbcae725fb28ae559cb/gst/tensor_decoder/tensordec.c#L582-L591
https://github.com/nnsuite/nnstreamer/blob/a6d3314c0ad3781bfa00cbbcae725fb28ae559cb/gst/tensor_filter/tensor_filter.c#L792-L802
https://github.com/nnsuite/nnstreamer/blob/a6d3314c0ad3781bfa00cbbcae725fb28ae559cb/gst/tensor_demux/gsttensordemux.c#L436-L451

Therefore, I just added g_assert here for just in case. In addition, using the combination of gst_tensor_info_get_size and gst_memory and gst_buffer_get_size (), it is possible to change this code to more secure way, if you want.